src/server/v2.0/handler/scan_all.go

```diff
@@ -0,0 +1,248 @@
+
+		if _, err := s.createOrUpdateScanAllSchedule(ctx, req.Schedule.Type, req.Schedule.Cron, nil); err != nil {
+			return s.SendError(ctx, err)
+		}
+	}
+
+	return operation.NewCreateScanAllScheduleCreated()
+}
+
+func (s *scanAllAPI) UpdateScanAllSchedule(ctx context.Context, params operation.UpdateScanAllScheduleParams) middleware.Responder {
+	req := params.Schedule
+
+	if req.Schedule.Type == ScheduleManual {
+		message := fmt.Sprintf("fail to update scan all schedule as wrong schedule type: %s", req.Schedule.Type)
+		return s.SendError(ctx, errors.BadRequestError(nil).WithMessage(message))
+	}
+
+	schedule, err := s.getScanAllSchedule(ctx)
+	if err != nil {
+		return s.SendError(ctx, err)
+	}
+
+	if req.Schedule.Type == ScheduleNone {
+		if schedule != nil {
+			err = s.scheduler.UnScheduleByID(ctx, schedule.ID)
+		}
+	} else {
+		_, err = s.createOrUpdateScanAllSchedule(ctx, req.Schedule.Type, req.Schedule.Cron, schedule)
+	}
+
+	if err != nil {
+		return s.SendError(ctx, err)
+	}
+
+	return operation.NewUpdateScanAllScheduleOK()
+}
+
+func (s *scanAllAPI) GetScanAllSchedule(ctx context.Context, params operation.GetScanAllScheduleParams) middleware.Responder {
+	schedule, err := s.getScanAllSchedule(ctx)
+	if err != nil {
+		return s.SendError(ctx, err)
+	}
+
+	return operation.NewGetScanAllScheduleOK().WithPayload(model.NewSchedule(schedule).ToSwagger())
+}
+
+func (s *scanAllAPI) GetLatestScanAllMetrics(ctx context.Context, params operation.GetLatestScanAllMetricsParams) middleware.Responder {
+	stats, err := s.getMetrics(ctx)
+	if err != nil {
+		return s.SendError(ctx, err)
+	}
+
+	return operation.NewGetLatestScanAllMetricsOK().WithPayload(stats)
+}
+
+func (s *scanAllAPI) GetLatestScheduledScanAllMetrics(ctx context.Context, params operation.GetLatestScheduledScanAllMetricsParams) middleware.Responder {
+	stats, err := s.getMetrics(ctx, task.ExecutionTriggerSchedule)
+	if err != nil {
+		return s.SendError(ctx, err)
+	}
+
+	return operation.NewGetLatestScanAllMetricsOK().WithPayload(stats)
+}
+
+func (s *scanAllAPI) createOrUpdateScanAllSchedule(ctx context.Context, cronType, cron string, previous *scheduler.Schedule) (int64, error) {
+	if previous != nil {
+		if cronType == previous.CRONType && cron == previous.CRON {
+			return previous.ID, nil
+		}
+
+		if err := s.scheduler.UnScheduleByID(ctx, previous.ID); err != nil {
+			return 0, err
+		}
+	}
+
+	return s.scheduler.Schedule(ctx, job.ImageScanAllJob, 0, cronType, cron, scan.ScanAllCallback, nil, nil)
+}
+
+func (s *scanAllAPI) getScanAllSchedule(ctx context.Context) (*scheduler.Schedule, error) {
+	query := q.New(q.KeyWords{"vendor_type": job.ImageScanAllJob})
+	schedules, err := s.scheduler.ListSchedules(ctx, query.First("-creation_time"))
+	if err != nil {
+		return nil, err
+	}
+
+	if len(schedules) > 1 {
+		return nil, fmt.Errorf("found more than one scheduled scan all job, please ensure that only one schedule left")
+	} else if len(schedules) == 0 {
+		return nil, nil
```

Comment:
return ErrNotFound?

Reply:
No need to return `ErrNotFound` here because get schedule API doesn't return the 404 when the schedule not found. It's really confusing.